TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -372,5 +372,113 @@ public LockTableStatement CreateLockTableStatement(CodeElementsParser.LockTableS
+        {
+            if (context.simpleAssignmentClause() != null)
+            {
+                return CreateSimpleAssignmentClause(context.simpleAssignmentClause());
+            }
+            else if (context.multipleAssignmentClause() != null)
+            { 
+                return CreateMultipleAssignmentClause(context.multipleAssignmentClause());
+            }
+            return null;
+        }
+
+        private Assignment CreateMultipleAssignmentClause(CodeElementsParser.MultipleAssignmentClauseContext context)
+        {
+            IList<SourceValue> values = null;
+            IList<TargetVariable> targets = context.targetVariable().Select(CreateTargetVariable).ToList();
+
+            if (context.sourceValueClause().sourceValueClauses().repeatedSourceValue() != null)
+            {
+                values = CreateRepeatedSourceValue(context.sourceValueClause().sourceValueClauses()
+                    .repeatedSourceValue());
+            }
+            else if (context.sourceValueClause().sourceValueClauses().sourceValue() != null)
+            {
+                values = new List<SourceValue>
+                {
+                    CreateSourceValue(context.sourceValueClause().sourceValueClauses().sourceValue())
+                };
+            }
+
+            return new Assignment(targets, values);
+        }
+
+        private Assignment CreateSimpleAssignmentClause(CodeElementsParser.SimpleAssignmentClauseContext context)
+        {
+            IList<TargetVariable> targets = new List<TargetVariable>();
+            IList<SourceValue> values = new List<SourceValue>();
+            if (context.targetVariable() != null)
+            {
+                targets.Add(new TargetVariable(CreateSqlVariable(context.targetVariable().sqlVariable())));
+            }
+            if (context.sourceValue() != null)
+            {
+                values.Add(CreateSourceValue(context.sourceValue()));
+            }
+            return new Assignment(targets, values);
+        }
+
+        private List<SourceValue> CreateRepeatedSourceValue(CodeElementsParser.RepeatedSourceValueContext context)
+        {
+            var sourceValues = context.sourceValue().Select(CreateSourceValue).ToList();
+            return sourceValues;
+        }
+
+        private TargetVariable CreateTargetVariable(CodeElementsParser.TargetVariableContext context)
+        {
+            return context.sqlVariable() != null ? new TargetVariable(CreateSqlVariable(context.sqlVariable())) : null;
+        }
+
+        private SourceValue CreateSourceValue(CodeElementsParser.SourceValueContext context)
+        {
+            if (context.sqlExpression() == null) return null;
+            var isDefault = context.SQL_DEFAULT() != null
+                ? new SyntaxProperty<bool>(true, ParseTreeUtils.GetFirstToken(context.SQL_DEFAULT()))
+                : null;
+            var expression = CreateSqlExpression(context.sqlExpression());
```

Comment:
Isn't context.sqlExpression() null when DEFAULT is used ? Then expression should be null and the method not called.